synadm/api.py

```diff
@@ -304,7 +304,7 @@ def generate_mxid(self, user_id):
         """
         if user_id is None:
             return None
-        elif re.match(r"@[\w\d\_\.\_\=\-\/]+:[\w\d]+", user_id):
+        elif re.match(r"@[-_/=\w\d\.]+:[-_.\w\d]+", user_id):
```

Comment:
localpart is fine now. I think serverpart should be `[-.w\d]+` (without the underscore, because it's not allowed in URL's, but maybe you found out something more about possibly allowed chars. It doesn't hurt too much if we keep the underscore in there I think :-)

I just read the docs again and I think even the dot does not need to be escaped with the backslash because it is inside a set `[abc]`, so either way, we shojuld be consistent how to put in the dot in both the local and the serverpart. Both ways are fine for me and will be understood by regex-aware coders.

Reply:
> I think serverpart should be [-.\w\d]+ 

Your are right, I will remove the underscore. 

Regarding the `.` in the server part, we should remove the escaping slash. The reason why I am added escaping slashed before every special character, is that I wanted to make absolutely sure that these characteres are only interpreted as plain text. However, I agree that the regex should be consistent, so I suggest that we remove slashes wherever possible.

Reply:
I just had a closer look again to the the servername matrix spec you posted a couple of days ago I think we should support everything that's stated there but we are on the right track already :-)

But as far as I can see it actually _is_ just what domain names and IP adresses are: 

```
This grammar is based on the standard for internet host names, as specified by [RFC1123, section 2.1](https://tools.ietf.org/html/rfc1123#page-13), with an extension for IPv6 literals.
```

So in short I think this should suffice: `[.-\w\d]`

Reply:
This is great! I fixed the regex accordingly in ed9b4f6e9115f1b4e09b94e3987706219d2b7335. But I changed the server part to `[-.\w\d]+` since the `-` must be the first character within the set, otherwise an error occurs.

Reply:
Ah yes, you are right, forgot that already 😀
And just now I also recall that \w includes \d already.
Besides that I think we are ready to merge 🎉

Reply:
I just figured out that `\w` also includes underscore characters as stated in the documentation (https://docs.python.org/3/howto/regex.html). Therefore I think that we can also remove the explicit underscore.

Reply:
Just removed the `\d` and the underscore `_` from the regex in [46427c9](https://github.com/JOJ0/synadm/pull/59/commits/46427c9df614a5570ab01e229b454c3618d32c51).

If there is nothing left, we are finally ready to merge.